packages/tao/src/compat/compat.ts

```diff
@@ -37,20 +50,32 @@ if (!patched) {
         `@angular-devkit/core/src/workspace/json/reader`,
       ]);
       const originalReadJsonWorkspace = readJsonUtils.readJsonWorkspace;
-      readJsonUtils.readJsonWorkspace = () => {
-        // Read our v1 workspace schema
-        const w = resolveOldFormatWithInlineProjects(
-          new Workspaces(appRootPath).readWorkspaceConfiguration()
-        );
-        // readJsonWorkspace actually has AST parsing + more, so we
-        // still need to call it rather than just return our file
-        return originalReadJsonWorkspace.apply(this, [
-          'workspace.json', // path name, doesn't matter
-          {
-            // second arg is a host, only method used is readFile
-            readFile: () => JSON.stringify(w),
-          },
-        ]);
+      readJsonUtils.readJsonWorkspace = (
+        path,
+        host: { readFile: (p) => Promise<string> }
+      ) => {
+        try {
+          return originalReadJsonWorkspace(path, host);
+        } catch {
```

Comment:
During schematic execution this will succeed, and reads from the host which is adapted inside ngcli-adapter.ts. During builder execution, this will fail w/o workspace.json present and fall back on the Workspaces class.